### PR TITLE
CHE-7872: Rework Git branch dialog filter to have focus from the whole window

### PR DESCRIPTION
**Pull request by vinokurig**
### What does this PR do?
Rework Git branch dialog filter to have focus from the whole window. This fixes bug when filter doesn't work after opening filter drop-down and after create, delete and rename branches.

### What issues does this PR fix or reference?
#7872 

Reworked Git branch dialog filter to have focus from the whole window.

#### Release Notes

N/A

#### Docs PR

N/A

### CODE REVIEW

File: ide/commons-gwt/src/main/java/org/eclipse/che/ide/ui/list/FilterableSimpleList.java [hunk 1]
```diff
@@ -75,6 +45,32 @@ private FilterableSimpleList(
...
+
+  /** Clear the filter and show all items. */
+  public void resetFilter() {
+    clearFilter();
+    filterChangedHandler.onFilterChanged("");
+    doFilter();
+  }
+
+  /** Add Backspace to the filter and update items according to new filter value. */
+  public void addBackspaceToFilter() {
```
Comment (tolusha) [inline ide/commons-gwt/src/main/java/org/eclipse/che/ide/ui/list/FilterableSimpleList.java]: removeLastCharacter

**Review by tolusha — APPROVED** [review]

**Comment by tolusha** [comment]
ci-test

File: ide/commons-gwt/src/main/java/org/eclipse/che/ide/ui/list/FilterableSimpleList.java (hunk 1 above)
Comment (vinokurig) [reply ide/commons-gwt/src/main/java/org/eclipse/che/ide/ui/list/FilterableSimpleList.java]: done

**Review by vzhukovs — APPROVED** [review]

**Comment by vinokurig** [comment]
All failed selenium tests where successfully passed locally

**Comment by tolusha** [comment]
Let's rerun one more time

**Comment by tolusha** [comment]
ci-test

**Review by vparfonov — APPROVED** [review]

**Pull request merged**